presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java

```diff
@@ -0,0 +1,206 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.facebook.presto.execution.SqlTask;
+import com.facebook.presto.execution.SqlTaskIoStats;
+import com.facebook.presto.execution.SqlTaskManager;
+import com.facebook.presto.execution.TaskInfo;
+import com.facebook.presto.execution.TaskStatus;
+import com.facebook.presto.memory.QueryContext;
+import com.facebook.presto.operator.TaskStats;
+import com.facebook.presto.spi.QueryId;
+import io.airlift.log.Logger;
+import io.airlift.stats.GarbageCollectionNotificationInfo;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.management.JMException;
+import javax.management.Notification;
+import javax.management.NotificationListener;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeData;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class GcStatusMonitor
+{
+    private final Logger log = Logger.get(GcStatusMonitor.class);
+    private final NotificationListener notificationListener = (notification, ignored) -> onNotification(notification);
+    private final SqlTaskManager sqlTaskManager;
+
+    @Inject
+    public GcStatusMonitor(
+            SqlTaskManager taskManager)
+    {
+        this.sqlTaskManager = taskManager;
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        for (GarbageCollectorMXBean mbean : ManagementFactory.getGarbageCollectorMXBeans()) {
+            ObjectName objectName = mbean.getObjectName();
+            try {
+                ManagementFactory.getPlatformMBeanServer().addNotificationListener(
+                        objectName,
+                        notificationListener,
+                        null,
+                        null);
+            }
+            catch (JMException e) {
+                throw new RuntimeException("Unable to add GC listener", e);
+            }
+        }
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        for (GarbageCollectorMXBean mbean : ManagementFactory.getGarbageCollectorMXBeans()) {
+            ObjectName objectName = mbean.getObjectName();
+            try {
+                ManagementFactory.getPlatformMBeanServer().removeNotificationListener(objectName, notificationListener);
+            }
+            catch (JMException ignored) {
+            }
+        }
+    }
+
+    private synchronized void onNotification(Notification notification)
+    {
+        if ("com.sun.management.gc.notification".equals(notification.getType())) {
```

Comment:
Please extract the `"com.sun.management.gc.notification"` into a static constant with a meaningful name, e.g.: `GC_NOTIFICATION_TYPE`